### PR TITLE
global: deposition jsonschema and mapping

### PR DESCRIPTION
**Pull request by krzysztof**
* Adds deposition jsonschema and elasticsearch mapping.

### CODE REVIEW

File: zenodo/modules/deposit/indexer.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    else:
+        json['_updated'] = record['_deposit']['submitted']
+
+    if 'recid' in record['_deposit'] and \
+            record['_deposit']['status'] == 'published':
+        recid = record['_deposit']['recid']
+        resolver = Resolver(pid_type='recid', object_type='rec',
+                            getter=Record.get_record)
+        rec_pid, rec = resolver.resolve(recid)
+        json['title'] = rec['title']  # TODO: map record to deposit here
```
Comment (lnielsen) [inline zenodo/modules/deposit/indexer.py]: You can probably do something along the lines of:

```python
json.clear()
json.update(rec)
```

Just the ``_deposit`` we have to take care with.

File: zenodo/modules/records/indexer.py [hunk 2]
```diff
@@ -30,13 +30,11 @@
 def indexer_receiver(sender, json=None, record=None, index=None,
                      **dummy_kwargs):
     """Connect to before_record_index signal to transform record for ES."""
+    if not index.startswith('records-'):
+        return
     # Inject timestamp into record.
     json['_created'] = record.created
```
Comment (lnielsen) [inline zenodo/modules/records/indexer.py]: I think we need these timestamps for grants and funders, but probably we should add these two lines to [Invenio-Indexer](https://github.com/inveniosoftware/invenio-indexer/blob/master/invenio_indexer/api.py#L297) by default.

File: zenodo/modules/deposit/jsonschemas/deposits/records/record-v1.0.0.json [hunk 3]
```diff
@@ -1,9 +1,761 @@
 {
-    "$schema": "http://json-schema.org/draft-04/schema#",
```
Comment (lnielsen) [inline zenodo/modules/deposit/jsonschemas/deposits/records/record-v1.0.0.json]: This part is superseded by #563

**Comment by krzysztof** [comment]
Depends on #569

**Comment by krzysztof** [comment]
Depends on #569

**Comment by krzysztof** [comment]
@lnielsen Should be review ready

File: zenodo/modules/deposit/indexer.py [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    :type json: `invenio_records.api.Record`
+    :param record: Indexed record.
+    :type record: `invenio_records.api.Record`
+    :param index: Elasticsearch index.
+    :type index: str
+    """
+    # Inject timestamp into record.
+    if not index.startswith('deposits-records-'):
+        return
+    json['_created'] = record.created
```
Comment (lnielsen) [inline zenodo/modules/deposit/indexer.py]: can be removed

File: zenodo/modules/deposit/indexer.py [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+    if record['_deposit']['status'] != 'published':
+        json['_updated'] = record.updated
+    else:
+        json['_updated'] = record['_deposit']['submitted']
+
+    if 'recid' in record['_deposit'] and \
+            record['_deposit']['status'] == 'published':
+        recid = record['_deposit']['recid']
```
Comment (lnielsen) [inline zenodo/modules/deposit/indexer.py]: ``record.fetch_published()`` perhaps works

**Comment by krzysztof** [comment]
Depends on #582

File: requirements.devel.txt [hunk 6]
```diff
@@ -27,3 +27,4 @@
 -e git+https://github.com/inveniosoftware/invenio-migrator.git#egg=invenio-migrator
 -e git+https://github.com/inveniosoftware/invenio-oauth2server.git#egg=invenio-oauth2server
 -e git+https://github.com/zenodo/zenodo-migrationkit.git#egg=zenodo-migrationkit
+-e git+https://github.com/zenodo/zenodo.git#egg=zenodo
```
Comment (lnielsen) [inline requirements.devel.txt]: I don't think this one should be here.

File: setup.py [hunk 7]
```diff
@@ -132,7 +132,6 @@
     'Pillow>=3.2.0',
     'python-slugify>=1.2.0',
     'raven<=5.1.0',
-    'zenodo-migrationkit>=1.0.0.dev20150000',
```
Comment (lnielsen) [inline setup.py]: 👍

File: zenodo/modules/deposit/indexer.py [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+    :type record: `invenio_records.api.Deposit`
+    :param index: Elasticsearch index name.
+    :type index: str
+    """
+    if not index.startswith('deposits-records-'):
+        return
+    if record['_deposit']['status'] == 'published':
+        pub_pid, pub_rec = record.fetch_published()
+        schema = json['$schema']
+        json['_deposit']['status'] = 'draft'  # FIXME: cannot clear otherwise
```
Comment (lnielsen) [inline zenodo/modules/deposit/indexer.py]: Fix in Invenio-Deposit or Zenodo?

Comment (krzysztof) [inline zenodo/modules/deposit/indexer.py]: Perhaps as a kwarg in Zenodo Deposit API: ``json.clear(force=True)`` ?

Comment (krzysztof) [inline zenodo/modules/deposit/indexer.py]: Removed the FIXME, I'd keep it as is for this PR

**Comment by lnielsen** [comment]
Cherry picked into a3226407adeeb7a6e4bd13ddd47e56f821ba9855

**Pull request closed without merging**